src/ApiPort.VisualStudio/ApiPortVSPackage.cs

```diff
@@ -100,8 +100,12 @@ protected override void Initialize()
             }
         }
 
-        public void ShowToolbar()
+        public async System.Threading.Tasks.Task ShowToolbarAsync()
         {
+            // Calls to UI elements should use the main task thread.
+            // https://blogs.msdn.microsoft.com/andrewarnottms/2014/05/07/asynchronous-and-multithreaded-programming-within-vs-using-the-joinabletaskfactory/
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Not sure if VS does stuff differently, but UI events are generally called on the UI thread (at least in WPF/UWP)